go/vt/topo/keyspace_routing_rules_lock.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topo
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"vitess.io/vitess/go/vt/log"
+)
+
+// KeyspaceRoutingRulesLock is a wrapper over TopoLock, to serialize updates to the keyspace routing rules.
+type KeyspaceRoutingRulesLock struct {
+	*TopoLock
+}
+
+func createTopoDirIfNeeded(ctx context.Context, ts *Server) error {
+	// We need the directory to exist since etcd (for example) will create a file there as part of its locking
+	// mechanism. So we create it if it doesn't exist. The file created below "lock" is just a dummy file.
+	// If no dummy file is specified, the directory doesn't get created (etcd behavior).
+	topoPath := path.Join(KeyspaceRoutingRulesPath, "lock")
```

Comment:
Nit, but `locks` is used for the other topo locks. Probably worth being consistent since it doesn't make much difference to us here. It's also more consistent with how the topo server implementations handle the locking using the prefix (a unique suffix is added for each lock).